adal/src/main/java/com/microsoft/aad/adal/TokenCacheItem.java

```diff
@@ -106,6 +111,9 @@ private TokenCacheItem(final String authority, final AuthenticationResult authen
         mRawIdToken = authenticationResult.getIdToken();
         mRefreshtoken = authenticationResult.getRefreshToken();
         mFamilyClientId = authenticationResult.getFamilyClientId();
+        if (authenticationResult.getExtendedExpiresOn() != null) {
```

Comment:
OK. Do you mean we should let the function who calling this constructor to verify the variables?